cmd/kube-controller-manager/app/plugins.go

```diff
@@ -61,18 +70,49 @@ func ProbeRecyclableVolumePlugins(flags VolumeConfigFlags) []volume.VolumePlugin
 		RecyclerTimeoutIncrement: flags.PersistentVolumeRecyclerIncrementTimeoutNFS,
 		RecyclerPodTemplate:      volume.NewPersistentVolumeRecyclerPodTemplate(),
 	}
-	if err := attemptToLoadRecycler(flags.PersistentVolumeRecyclerPodTemplateFilePathNFS, &nfsConfig); err != nil {
+	if err := AttemptToLoadRecycler(flags.PersistentVolumeRecyclerPodTemplateFilePathNFS, &nfsConfig); err != nil {
 		glog.Fatalf("Could not create NFS recycler pod from file %s: %+v", flags.PersistentVolumeRecyclerPodTemplateFilePathNFS, err)
 	}
 	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(nfsConfig)...)
 
+	allPlugins = append(allPlugins, aws_ebs.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, gce_pd.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, cinder.ProbeVolumePlugins()...)
+
 	return allPlugins
 }
 
-// attemptToLoadRecycler tries decoding a pod from a filepath for use as a recycler for a volume.
+// NewVolumeProvisioner returns a volume provisioner to use when running in a cloud or development environment.
+// The beta implementation of provisioning allows 1 implied provisioner per cloud, until we allow configuration of many.
+// We explicitly map clouds to volume plugins here which allows us to configure many later without backwards compatibility issues.
+// Not all cloudproviders have provisioning capability, which is the reason for the bool in the return to tell the caller to expect one or not.
+func NewVolumeProvisioner(plugins []volume.VolumePlugin, cloud cloudprovider.Interface, flags VolumeConfigFlags) (volume.ProvisionableVolumePlugin, error) {
+	switch {
+	case cloud != nil && aws.ProviderName == cloud.ProviderName():
+		return getProvisionablePluginFromVolumePlugins(aws_ebs.ProbeVolumePlugins())
+	case cloud != nil && gce.ProviderName == cloud.ProviderName():
+		return getProvisionablePluginFromVolumePlugins(gce_pd.ProbeVolumePlugins())
```

Comment:
Why not just remove those cloud providers for now. And then add them back in @jsafrane's PRs?